File: tk-content-generator/scripts/build_content_brief.js
#!/usr/bin/env node
const fs = require("fs");

function parseCsv(text) {
  const lines = text.trim().split(/\r?\n/);
  const headers = parseCsvLine(lines.shift()).map((h) => h.trim());
  return lines.filter(Boolean).map((line) => {
    const values = parseCsvLine(line).map((v) => v.trim());
    return Object.fromEntries(headers.map((h, i) => [h, values[i] ?? ""]));
  });
}

function parseCsvLine(line) {
  const values = [];
  let current = "";
  let quoted = false;
  for (let i = 0; i < line.length; i += 1) {
    const char = line[i];
    const next = line[i + 1];
    if (char === '"' && quoted && next === '"') {
      current += '"';
      i += 1;
    } else if (char === '"') {
      quoted = !quoted;
    } else if (char === "," && !quoted) {
      values.push(current);
      current = "";
    } else {
      current += char;
    }
  }
  values.push(current);
  return values;
}

function toCsv(rows) {
  const headers = [...new Set(rows.flatMap((row) => Object.keys(row)))];
  const escape = (value) => {
    const text = String(value ?? "");
    return /[",\n]/.test(text) ? `"${text.replace(/"/g, '""')}"` : text;
  };
  return [headers.join(","), ...rows.map((row) => headers.map((h) => escape(row[h])).join(","))].join("\n");
}

function pick(row, key, fallback) {
  return (row[key] || fallback).trim();
}

function build(row) {
  const product = pick(row, "product", "this product");
  const audience = pick(row, "audience", "busy shoppers");
  const pain = pick(row, "pain", "a common daily problem");
  const benefit = pick(row, "benefit", "make the task easier");
  const proof = pick(row, "proof", "the demo shows the difference");
  const offer = pick(row, "offer", "check the product page");
  const category = pick(row, "category", "shopping").replace(/\s+/g, "");
  const country = pick(row, "country", "targetmarket").replace(/\s+/g, "");
  return {
    ...row,
    hook: `If ${pain} keeps happening, watch what ${product} does.`,
    demo: `Show ${product} solving ${pain} in one clear before-and-after shot.`,
    proof,
    cta: offer,
    shot_list: "hook close-up | pain scene | product demo | result comparison | CTA product shot",
    voiceover: `I got ${product} because ${pain}. Here is the quick test. It helps ${audience} ${benefit}. ${proof}. ${offer}.`,
    captions: `Still dealing with ${pain}? / Testing ${product} / Made for ${audience} / ${benefit} / ${offer}`,
    title: `I tested ${product} for ${pain}`,
    hashtags: `#${category} #${country} #tiktokshop #finds #musthave`,
  };
}

const [input, output] = process.argv.slice(2);
if (!input || !output) {
  console.error("Usage: node build_content_brief.js input.csv output.csv");
  process.exit(1);
}

fs.writeFileSync(output, toCsv(parseCsv(fs.readFileSync(input, "utf8")).map(build)), "utf8");

File: tk-content-generator/scripts/build_content_brief.py
#!/usr/bin/env python3
import argparse
import csv
from pathlib import Path


def value(row, key, default=""):
    return (row.get(key) or default).strip()


def build_brief(row):
    product = value(row, "product", "this product")
    audience = value(row, "audience", "busy shoppers")
    pain = value(row, "pain", "a common daily problem")
    benefit = value(row, "benefit", "make the task easier")
    proof = value(row, "proof", "the demo shows the difference")
    offer = value(row, "offer", "check the product page")
    category = value(row, "category", "shopping")
    country = value(row, "country", "target market")

    hook = f"If {pain} keeps happening, watch what {product} does."
    demo = f"Show {product} solving {pain} in one clear before-and-after shot."
    voiceover = (
        f"I got {product} because {pain}. Here is the quick test. "
        f"It helps {audience} {benefit}. {proof}. {offer}."
    )
    captions = [
        f"Still dealing with {pain}?",
        f"Testing {product}",
        f"Made for {audience}",
        f"{benefit}",
        offer,
    ]
    hashtags = [
        f"#{category.replace(' ', '')}",
        f"#{country.replace(' ', '')}",
        "#tiktokshop",
        "#finds",
        "#musthave",
    ]

    return {
        **row,
        "hook": hook,
        "pain": pain,
        "demo": demo,
        "proof": proof,
        "cta": offer,
        "shot_list": "hook close-up | pain scene | product demo | result comparison | CTA product shot",
        "voiceover": voiceover,
        "captions": " / ".join(captions),
        "title": f"I tested {product} for {pain}",
        "hashtags": " ".join(hashtags),
    }


def main():
    parser = argparse.ArgumentParser(description="Build TikTok content briefs from product rows.")
    parser.add_argument("input_csv", type=Path)
    parser.add_argument("output_csv", type=Path)
    args = parser.parse_args()

    with args.input_csv.open("r", newline="", encoding="utf-8-sig") as source:
        rows = list(csv.DictReader(source))

    briefs = [build_brief(row) for row in rows]
    fieldnames = list(dict.fromkeys([key for row in briefs for key in row.keys()]))

    with args.output_csv.open("w", newline="", encoding="utf-8") as target:
        writer = csv.DictWriter(target, fieldnames=fieldnames)
        writer.writeheader()
        writer.writerows(briefs)


if __name__ == "__main__":
    main()
